### ISSUE TITLE
Frontend: Switching between networks triggers a wallet disconnect

### ISSUE THREAD

**Issue by GabrielBuragev** [beamer-bridge/beamer#1118]
The implementation of the internal wallet disconnect detection somehow breaks the network switching flow. 
When a network switch is being innitiated the `Eip1193Provider` emits a `disconnect` event for some reason.
We need to rethink the implementation here and come up with another solution.

![wallet-disconnect](https://user-images.githubusercontent.com/18466333/194840329-2beb984c-4e59-4e19-ac06-f75f04fad5e9.gif)

## Accaptence criteria
- [ ] Make sure problem does not happen for all wallets that we support

Related issue: #692 
Blame PR: https://github.com/beamer-bridge/beamer/pull/1087

**Comment by manuelwedler** [comment]
I've been investigating this issue a little.. my findings so far: 

- I cannot reproduce it with every network switch, it only happens around 50 % or maybe less of the times. 
- The `.isConnected()` method on the external provider returns false for a very short time period, even though MetaMask is still connected. This is when it also emits the `disconnect` event. After that time period the `.isConnected()` method returns true again.
- There is this issue on MetaMask describing the problem: https://github.com/MetaMask/metamask-extension/issues/13375 , so this seems to be a bug on their side.

**Comment by andfletcher** [comment]
@GabrielBuragev mentioned that this might not be fixed and we should double check.

**Comment by GabrielBuragev** [comment]
While testing the wallet providers, i noticed that the speed of the RPC provider that is used inside metamask/dapp has an impact.  

With a fast RPC provider i cannot reproduce this issue. However, when i try it with a slow RPC provider it is there...

**Comment by manuelwedler** [comment]
I could not really reproduce the problem. The only times it happened for me was when the network was not added to metamask yet. In this case, I think out timer simply runs out before the user clicks to add the network which fires the disconnect event then. 

I had a look into the MetaMask code base, but it's so big and would take time to get into it. I don't think it's worth the effort to fix it on their side.

At the moment we solve the issue by setting a timeout of 1 second. During this period we ignore the disconnect event. One simple solution to the slower rpcs you mentioned could be to increase our timeout period to a few more seconds. Still, this is not optimal.

**Comment by GabrielBuragev** [comment]
Wagmi library has a fix implemented in their metamask connector. Maybe a good reference:
https://github.com/wagmi-dev/references/blob/main/packages/connectors/src/metaMask.ts

**Comment by manuelwedler** [comment]
So wagmi tries to fix it by having a flag that is turned to false once the disconnect event was thrown (at the moment we change the flag after a timeout). Would look like this in our code:
```
disconnect(): void {
   if (this.isSwitchingNetwork) {
         this.isSwitchingNetwork = false;
         return;
   }
   //...
}
```
The issue with this approach is that the disconnect is not always thrown. In case none is thrown, the flag is not turned to false again. That means that we would not catch a real disconnect event. So when a user clicks disconnect inside MetaMask we would still show the connected state in our dapp. Tested it like this and I run into the issue as described. Really wonder if integrating wagmi actually would have fixed the issue.

I tried one thing, namely just removing our listener to the disconnect event. Instead the accounts changed event can be used to catch the case that a user clicks disconnect inside MetaMask. So far it looks like it is working fine.
